File: ProtoSAS/src/main/java/dd/sas/pipeline/calculation/structure/flows/associativeflows/RelationFlow.java
package dd.sas.pipeline.calculation.structure.flows.associativeflows;

import dd.sas.pipeline.worldmodel.Relation;

import java.util.List;

/**
 * Created by Sergey on 18.09.2016.
 */
public abstract class RelationFlow<R extends Relation> implements RelationFlowI<R> {

    public boolean checkLInFlow(List list) {
        return (list != null && !list.isEmpty());
    }
}

File: ProtoSASOnScala/src/main/scala/dd/sas/pipeline/calculation/structures/flows/associativeflows/RelationFlow.scala
package dd.sas.pipeline.calculation.structures.flows.associativeflows

import dd.sas.pipeline.calculation.structures.flows.Flow
import dd.sas.pipeline.calculation.structures.nodes.Node
import dd.sas.pipeline.worldmodel.{Relation, WorldObject}

/**
  * Created by Sergey on 12.10.2016.
  */
abstract class RelationFlow[D <: WorldObject, R <: WorldObject, S <: Relation[D, R]]
(node: Node) extends Flow(node) {
  def apply(): List[S]
}
